core/src/main/java/com/linecorp/armeria/internal/metric/CaffeineMetrics.java

```diff
@@ -0,0 +1,208 @@
+
+    /**
+     * Creates a new instance with the specified {@link Metrics}, {@link MetricKey} and {@link Cache}.
+     */
+    public CaffeineMetrics(Metrics parent, MetricKey key, Cache<?, ?> cache) {
+        this(parent, key, cache, Ticker.systemTicker());
+    }
+
+    /**
+     * Creates a new instance with the specified {@link Metrics}, {@link MetricKey} and {@link Cache}.
+     */
+    public CaffeineMetrics(Metrics parent, MetricKey key, Cache<?, ?> cache, Ticker ticker) {
+        this.key = requireNonNull(key, "key");
+        this.cache = requireNonNull(cache, "cache");
+        this.ticker = requireNonNull(ticker, "ticker");
+
+        updateCacheStats(true);
+
+        hit = parent.register(new CacheStatGauge(
+                key.append("hit"), COUNT_CUMULATIVE, "the number of cache hits",
+                () -> cacheStats.hitCount()));
+        miss = parent.register(new CacheStatGauge(
+                key.append("miss"), COUNT_CUMULATIVE, "the number of cache misses",
+                () -> cacheStats.missCount()));
+
+        total = parent.register(new CacheStatGauge(
+                key.append("total"), COUNT_CUMULATIVE, "the number of cache hits and misses",
+                () -> cacheStats.requestCount()));
+
+        loadSuccess = parent.register(new CacheStatGauge(
+                key.append("loadSuccess"), COUNT_CUMULATIVE, "the number of successful cache loads",
+                () -> cacheStats.loadSuccessCount()));
+        loadFailure = parent.register(new CacheStatGauge(
+                key.append("loadFailure"), COUNT_CUMULATIVE, "the number of failed cache loads",
+                () -> cacheStats.loadFailureCount()));
+        loadTotal = parent.register(new CacheStatGauge(
+                key.append("loadTotal"), COUNT_CUMULATIVE, "the number of cache loads (success and failure)",
+                () -> cacheStats.loadCount()));
+
+        loadDuration = parent.register(new CacheStatGauge(
+                key.append("loadDuration"), NANOSECONDS_CUMULATIVE, "the total load time (success and failure)",
+                () -> cacheStats.totalLoadTime()));
+
+        eviction = parent.register(new CacheStatGauge(
+                key.append("eviction"), COUNT_CUMULATIVE, "the number of evicted entries",
+                () -> cacheStats.evictionCount()));
+        evictionWeight = parent.register(new CacheStatGauge(
+                key.append("evictionWeight"), COUNT_CUMULATIVE, "the sum of weight of evicted entries",
+                () -> cacheStats.evictionWeight()));
+
+        estimatedSizeGauge = parent.register(new CacheStatGauge(
+                key.append("estimatedSize"), COUNT, "the approximate number of entries",
+                () -> estimatedSize));
+    }
+
+    private void updateCacheStats() {
+        updateCacheStats(false);
+    }
+
+    private void updateCacheStats(boolean force) {
+        final long currentTimeNanos = ticker.read();
+        if (!force) {
```

Comment:
nit: Could we remove `force` parameter? `lastStatsUpdateTime` is initialized to 0, so it's not necessary I think.